scripts/gitlab/build_and_test.sh

```diff
@@ -27,39 +30,86 @@ then
     build_root=$(pwd)
 fi
 
-compiler=${COMPILER:-""}
-if [[ -z ${compiler} ]]
+conf=${CONFIGURATION:-""}
+if [[ -z ${conf} ]]
 then
-    echo "COMPILER is undefined... aborting" && exit 1
+    echo "CONFIGURATION is undefined... aborting" exit 1
 fi
 
 project_dir="$(pwd)"
-build_dir="${build_root}/build_${sys_type}_${compiler}"
+build_dir="${build_root}/build_${sys_type}_${conf}"
+install_dir="${build_root}/install_${sys_type}_${conf}"
 option=${1:-""}
 
 # Build
 if [[ "${option}" != "--test-only" ]]
 then
+    # 'conf' = toolchain__tuning
+    # 'host_config' = filter__tuning
+    #   where 'filter' can represent several toolchains
+    #   like <nvcc_10_gcc_X> covers any gcc paired with nvcc10
+    # 'toolchain' is a unique set of tools, and 'tuning' allows to have
+    # several configurations for this set, like <omptarget>.
+
+    echo "--- Configuration to match :"
+    echo "* ${conf}"
+
+    toolchain=${conf/__*/}
+    tuning=${conf/${toolchain}/}
+
+    # Find project host_configs matching the configuration
+    host_configs="$(ls host-configs/${sys_type}/ | grep "\.cmake$")"
+    echo "--- Available host_configs"
+    echo "${host_configs}"
+
+    match_count=0
+    host_config=""
+
+    # Translate file names into pattern to match the host_config
+    echo "--- Patterns"
+    for hc in ${host_configs}
+    do
+        pattern="${hc//X/.*}"
+        pattern="${pattern/.cmake/}"
+        echo "${pattern}"
+
+        if [[ -n "${tuning}" && ! "${pattern}" =~ .*${tuning}$ ]]
+        then
+            continue
+        fi
+
+        if [[ "${conf}" =~ ^${pattern}$ ]]
+        then
+            (( ++match_count ))
+            host_config="${hc}"
+            echo "-> Found Project Conf : ${host_config}"
+        fi
+    done
+
+    if (( match_count > 1 )) || (( match_count == 0 ))
+    then
+        echo "ERROR : none or multiple match(s) ..."
+        exit 1
+    fi
+
     # If building, then delete everything first
     rm -rf ${build_dir} && mkdir -p ${build_dir} && cd ${build_dir}
 
-    conf_suffix="host-configs/${sys_type}/${compiler}.cmake"
-
-    generic_conf="${project_dir}/.radiuss-ci/gitlab/conf/${conf_suffix}"
+    generic_conf="${project_dir}/.radiuss-ci/gitlab/conf/host-configs/${sys_type}/${toolchain}.cmake"
     if [[ ! -f ${generic_conf} ]]
     then
         echo "ERROR: Host-config file ${generic_conf} does not exist" && exit 1
     fi
 
-    umpire_conf="${project_dir}/${conf_suffix}"
-    if [[ ! -f ${umpire_conf} ]]
+    project_conf="${project_dir}/host-configs/${sys_type}/${host_config}"
+    if [[ ! -f ${project_conf} ]]
     then
-        echo "ERROR: Host-config file ${umpire_conf} does not exist" && exit 1
+        echo "ERROR: Host-config file ${project_conf} does not exist" && exit 1
```

Comment:
Can we make the existence of a project configuration file optional?  Right now, we have a bunch of empty project configuration files.

Reply:
Let’s say all the project host-config files for gcc* are empty. Then the filter mechanism allows you to create only one, named `gcc_X.cmake` for all the matching toolchain.
So there should be much less empty files needed.